src/main/java/ui/MenuControl.java

```diff
@@ -87,7 +87,35 @@ private Menu createFileMenu() {
         return file;
 
     }
-    
+
+    private void onBoardNew() {
+        logger.info("Menu: Boards > New");
+
+        if (confirmNewBoardCreation()) {
+            panels.closeAllPanels();
+            onBoardSaveAs();
+        } else {
+            logger.info("New board creation cancelled");
+        }
+    }
+
+    private boolean confirmNewBoardCreation() {
```

Comment:
A better name that captures both boolean return and the use of dialog: `isNewBoardCreationDialogConfirmed`